S/Siconos/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Siconos"
+version = v"4.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/siconos/siconos.git", "5b5a685c4f616aa514b0da1d0bd96009b1ad896f")
+    DirectorySource("./bundled/")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd siconos/
+mkdir build
+cd build/
+python3 -m pip install wheel packaging
+cmake -DUSER_OPTIONS_FILE=../../only_numerics.cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DWITH_FCLIB=0 -DWITH_PYTHON_WRAPPER=0 -DWITH_CXX=0 ..
```

Comment:
You specify the `WITH_CXX`, `WITH_PYTHON_WRAPPER` and `WITH_FCLIB` parts in the custom options file already, so do they really need to be passed on the command line as well?

Also, the values of `WITH_FCLIB`  and `WITH_CXX` here are different from the ones in the `only_numerics.cmake` file in the tree, so these will take priority over those, should you just update the file with these values?